osu.Game.Rulesets.Mania/Mods/ManiaModConstantSpeed.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mania.Objects;
+using osu.Game.Rulesets.Mania.UI;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.UI;
+
+namespace osu.Game.Rulesets.Mania.Mods
+{
+    public class ManiaModConstantSpeed : Mod, IApplicableToDrawableRuleset<ManiaHitObject>
+    {
+        public override string Name => "Constant Speed";
+
+        public override string Acronym => "CS";
+
+        public override double ScoreMultiplier => 1;
+
+        public override string Description => "No more tricky speed changes!";
+
+        public override IconUsage? Icon => FontAwesome.Solid.Equals;
+
+        public override ModType Type => ModType.Conversion;
+
+        public override bool Ranked => false;
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<ManiaHitObject> drawableRuleset)
+        {
+            var maniaRuleset = (DrawableManiaRuleset)drawableRuleset;
+            maniaRuleset.ScrollMethod = ScrollVisualisationMethod.Constant;
```

Comment:
So this is definitely a lot simpler than I expected and than what I was going for when I tried this, and it *works*, but I'm not sure how to feel about it - mostly because this working relies on the fact that `DrawableRuleset`'s BDL is called before `DrawableScrollingRuleset`'s, and therefore the latter - which initialises the immutable scrolling algorithm that then gets cached in `IScrollingInfo` - reads the correct value which was overridden "just in time".

While that seems like a sane way for BDL to work in general, and as such this solution may be fine, I'd maybe leave a comment explaining this at least? It's not a straightforward detail.

Also would maybe give writing a test a go? It would probably be pretty simplistic (resolve `IScrollingInfo` from the drawable ruleset and inspect its type), but I feel like this sort of solution is prone to accidental breakage from framework changes...

Reply:
> accidental breakage from framework changes

What sort of framework changes are you expecting to break this? Reversing the call order of `load()`? I'm pretty sure the game wouldn't even start up if that were the case.

Or by 'framework' you mean how the game calls this mod from the base `DrawableRuleset`? Sure - we've made no guarantees as to when mods get called in-relation to `load()`, but this isn't something that `DrawableRuleset` supports by itself anyway (read below) - it's just something that works for mania.

> publically settable property is useless

It's a publicly-settable property only for mania's drawable ruleset. I agree that in general the virtual property is useless post-`load()`, but that's an issue for another day and a breaking change in general. The algorithm should eventually be mutable but I noticed your mentioning in the superseded PR that "I don't deem this pull important enough to spend time on doing things properly", so I didn't spend the time to go through that and left it as an MVP just for mania. There's probably SHOC changes required to go down the 'proper' path.

And before going down that path, I don't think there's much point spending time on this. If it breaks it breaks - we can agree that the `VisualisationMethod` property should be fully mutable in the future, but this is very localised to mania. If it breaks then we'll do that and incur a breaking change for scrolling rulesets, but the mod implementation here won't change much.

Reply:
I've added a test but it's not testing correctness or anything other than the algorithm getting set in the first place.

Reply:
Everything you said was fair. It just wasn't stated that this was a somewhat temporary solution, which makes me more ok with this pull.

Reply:
If we wanted to be extra safe, the field could be made into a property that throws if it is changed post load?

Reply:
I wouldn't be against that.

Reply:
I can't tell if I'm being trolled right now... It's local to mania...

Regardless, I don't see the point. It's still an "invalid" operation if the code that calls `IApplicableToDrawableRuleset` is moved to, say, `LoadAsyncComplete` where it is still in the "Ready" state. The test that I added will be the only thing to catch that case.
Like I don't know what's being guarded against or what here, and this is all sounding like a waste of time.

Reply:
Unless I'm missing something adding the check would've been less words than your reply, but I'm fine either way. In general we try to avoid flimsy fields like this elsewhere in the code base, though.